src/harness/sas.py

```diff
@@ -332,3 +342,113 @@ def _GetDefaultAdminSSLCertPath(self):
+  """Test Harness acting as a Http Server to receive Pull/GET
+   Requests from SAS Under Test
+   parameters: is the dict variable for different types of dump objects.
+   schema of parameters: { "full_activity" : dict to store full activity dump objects,
+   "cbsd_activity": dict to store cbsd activity dump objects,
+   "PPA_activity" : dict to store ppa activity dump objects,
+   "esc_activity" : dict to store esc activity dump objects
+   }
+   schema of full_activity {full activity dump object}
+   schema of cbsd_activity {"id": {cbsd activity dump object}}
+   schema of PPA_activity {"id": {ppa activity dump object}}
+   schema of esc_activity {"id": {esc activity dump object}}
+
+  """
+  def __init__(self, server_details):
+    self._server_details = server_details
+    self.parameters = {}
+
+
+  def _handleRequest(self, getSetupParameters):
+    return lambda *args: HttpServerHandler(getSetupParameters, *args)
+
+  def _getSetupParameters(self):
+    return self.parameters
+
+
+  def setupFullActivity(self, full_activity_dump=None):
+    self.parameters['full_activity'] = full_activity_dump
+
+  def setupCbsdActivity(self, unique_id, Cbsd_activity_dump = None):
+    self.parameters['cbsd_activity'][unique_id] = Cbsd_activity_dump
+
+  def setupPPAActivity(self, unique_id, PPA_activity_dump = None):
+    self.parameters['PPA_activity'][unique_id] = PPA_activity_dump
+
+  def setupEscActivity(self,unique_id, esc_activity_dump = None):
+    self.parameter['esc_activity'][unique_id] = esc_activity_dump
+
+  def getBaseUrl(self):
+    return 'https://%s:%d' % (self._server_details['baseUrl'], self._server_details['port'])
+
+  def StartServer(self, ca_cert):
+    request_handler = self._handleRequest(self._getSetupParameters)
+    self.server = HTTPServer((self._server_details['baseUrl'], self._server_details['port']),request_handler)
+    cert_file = 'server.cert'
+    key_file = 'server.key'
+    ciphers = ['AES128-GCM-SHA256', 'AES256-GCM-SHA384', 'ECDHE-RSA-AES128-GCM-SHA256', 'ECDHE-ECDSA-AES256-GCM-SHA384', 'ECDHE-RSA-AES128-GCM-SHA256']
+    self.server.socket = ssl.wrap_socket(
+      self.server.socket,
+      certfile=cert_file,
+      keyfile=key_file,
+      ca_certs=ca_cert,
+      cert_reqs=ssl.CERT_REQUIRED,
+      ssl_version=ssl.PROTOCOL_TLSv1_2,
+      ciphers=':'.join(ciphers),
+      server_side=True
+      )
+    logging.info('Started Test Harness Server at %s' % self.getBaseUrl())
+    # Start Server
+    self.server.serve_forever()
+
+  def StopServer(self):
+    logging.info('Stopped Test Harness Server')
+    self.server.shutdown()
+
+class HttpServerHandler(BaseHTTPRequestHandler):
+  def __init__(self, getSetupParameters, *args):
+    self._parameters = getSetupParameters()
+    BaseHTTPRequestHandler.__init__(self, *args)
+
+  def do_GET(self):
+    """Handles Pull/GET Request and returns Path of the Request to callback Method"""
+    if "dump" in self.path:
```

Comment:
I think this check needs to be stricter. If I request $BASE_URL/dump_something, the request would succeed even though I should receive HTTP 404.